File: gmao-frontend/src/types/team.ts
export interface TeamMemberSummary {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
}

export interface TeamTagSummary {
  id: number;
  name: string;
  color: string;
}

export interface Team {
  id: number;
  name: string;
  description: string | null;
  members: TeamMemberSummary[];
  tags: TeamTagSummary[];
  createdAt: string;
  updatedAt: string;
}

export interface TeamRequest {
  name: string;
  description: string;
  memberIds: number[];
  tagIds: number[];
}

File: gmao-backend/src/main/java/gmao_backend/team/Team.java
package com.gmao.gmao_backend.team;

import com.gmao.gmao_backend.tag.Tag;
import com.gmao.gmao_backend.user.User;

import jakarta.persistence.*;
import lombok.*;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "teams")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Team {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 255)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @ManyToMany(mappedBy = "teams", fetch = FetchType.LAZY)
    @Builder.Default
    private Set<User> members = new HashSet<>();

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(
            name = "team_tags",
            joinColumns = @JoinColumn(name = "team_id"),
            inverseJoinColumns = @JoinColumn(name = "tag_id")
    )
    @Builder.Default
    private Set<Tag> tags = new HashSet<>();

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
}
